plugins/LilyPad/LilyPad.cpp

```diff
@@ -1195,15 +1188,15 @@ u8 CALLBACK PADpoll(u8 value) {
 				for (i = 0; i<8; i++) {
 					b2 -= (sum->buttons[i+4] > 0) << i;
 				}
+				for (i = 4; i<8; i++) {
+					b1 -= (sum->buttons[i+8] > 0) << i;
+				}
+
 				if (config.padConfigs[query.port][query.slot].type == GuitarPad && !config.GH2) {
-					sum->sticks[0].horiz = -255;
+					sum->buttons[15] = 255;
```

Comment:
I like the alternative better, but as I don't have a GH game to test, I'll go with your first suggestion. :)